src/utils.ts

```diff
@@ -208,9 +208,12 @@ export async function lookupAddress(address: string, config: Config): Promise<Pr
   try {
 
     const ens = await resolveEnsProfile(address, config);
-    if (ens) profile.ens = ens;
-    const idx = await resolveIdxProfile(formatCAIP10Address(address, "eip155", 3), config);
-    if (idx) profile.idx = idx;
+    const idx = await resolveIdxProfile(formatCAIP10Address(address, "eip155", config.network.chainId), config);
+    Promise.allSettled([ens, idx]).then(([ens, idx]) => {
+      profile.ens = ens.status == "fulfilled" ? ens.value : null;
+      profile.idx = idx.status == "fulfilled" ? idx.value : null;
+    });
```

Comment:
You have to await this promise, otherwise you're returning before it completes?

Eg. `let [ens, idx] = await Promise.allSettled(...)`

Reply:
I also believe you can simplify this, as `ens.value` will be `undefined` if the promise is rejected, so you don't need the coniditional and you can remove `null` from the type.

Reply:
> I also believe you can simplify this, as `ens.value` will be `undefined` if the promise is rejected, so you don't need the conditional and you can remove `null`s from the Profile type.

The issue is that when a `PromiseRejectedResult` is returned there is no `ens.value` and this doesn't work for the Typescript compiler.
However we could store the result from the `Promise.allSettled` function and let the UI look if the promise has been resolved or not.

Reply:
> I also believe you can simplify this, as `ens.value` will be `undefined` if the promise is rejected, so you don't need the conditional and you can remove `null`s from the Profile type.

I see a problem here since if the promise is rejected it has no property `value`, so the type checking fails.
However we could return the result of Promise.allSettled and let the components decide how they read the settled or rejected promise.

Reply:
Ah true, that's unfortunate. I think your solution could work though.

Reply:
8397b7876769a943e9df4d8653014a0c511b2b2a fixes this